src/iot.c

```diff
@@ -250,20 +251,15 @@ void shadow_thumb_upload_delta_handler(const char *json_str, uint32_t json_str_l
     }
     INFO("Delta - shadow state delta update: download_url %s\n", json_string_value(download_url));
 
-    // since we don't yet support grabbing actual thumbnails from the camera, get a random picture off the Internet
-    const char *thumbnail_path = "thumb.jpg";
-    if (download_file("http://lorempixel.com/200/200/nature/", thumbnail_path) != SUCCESS) {
-      thumbnail_path = "sample/snowdino.jpg";
-    }
+    IoT_Error_t rc = 0;
 
-    // upload to s3
-    FILE *fp = fopen(thumbnail_path, "rb");
-    if (!fp) {
-      ERROR("error: thumb fopen failed\n");
-      return;
+    // upload stream snapshot to s3
+    void *snapshot_data = NULL;
+    size_t snapshot_size = gst_get_jpeg_snapshot(ctx->gst, &snapshot_data);
+    if (snapshot_data != NULL && snapshot_size != 0) {
```

Comment:
maybe some error reporting could go here? perhaps report back to AWS even (like cloudwatch logs)

Reply:
This condition should only arise if GStreamer pipeline is completely dead; which would also be nice to check for and report back to the mothership. I'll have to think how to handle error reporting best (it would perhaps be nice to keep it within AWS IoT/MQTT protocol so there are no additional dependencies like full AWS SDK)